query/language/redefine.feature

```diff
@@ -448,34 +504,88 @@ Feature: TypeQL Redefine Query
       """
 
 
-  Scenario: can redefine relation type's owns ordering
+  Scenario: can redefine relation type's relates specialise, cannot redefine unchanged
+    Given typeql schema query
+      """
+      define part-time-employment relates part-time-role as empty-role;
+      """
+    Given transaction commits
+
+    Given connection open schema transaction for database: typedb
     Then typeql schema query; fails
       """
-      define relation employment owns start-date[];
+      define part-time-employment relates part-time-role as empty-abstract-role;
+      """
+    When transaction closes
+
+    When connection open schema transaction for database: typedb
+    When typeql schema query
+      """
+      redefine part-time-employment relates part-time-role as empty-abstract-role;
+      """
+    Then transaction commits
+
+    # TODO: Uncomment when querying for specialise works. Now it returns every `$_ relates $x`.
```

Comment:
TODO kind number four: `$A relates $B as $C` is not implemented. It has to be done with ValueTypes (TODO kind number two) as the first priority for match finishing.